pkg/fileio/file_io_test.go

```diff
@@ -0,0 +1,144 @@
+package fileio
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteFile(t *testing.T) {
+	const tmpDir = "write-test"
```

Comment:
Nit: I prefer to prefix temp directories with some indication of who created them (so something like `eib-write-test-`), but that's just a personal style.